### PR TITLE
Feature/htmlmap

### PR DESCRIPTION
**Pull request by mastoj**
The html map feature expanded somewhat. This adds the following concepts and changes: 

* Producers - async actions that produces messages to the application triggered outside of the application, a timer is a simple example.
* Actions - is async actions triggered from the update function.
* The state of the application has been simplified
* Event handling has been simplified, mostly due to the large amount of event types in js that is a mess. Now there is only one event handling type, and it is up to the user to define how the event should be decoded. In time we could add helpers for the most common scenarios.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
Nice! 👏 

Just one thing: it seems `samples/browser/lsystem/lsystem.fsproj` sneaked into the PR (probably it was modified by Paket or Forge). Could you please remove it, please?

File: samples/browser/virtualdom/virtualdom.fsx [hunk 1]
```diff
@@ -117,19 +117,80 @@ You define a `svg` the same way as you do with any other html element.
...
+
+type NestedModel = { Top: int; Bottom: int}
+
+type NestedAction = 
+    | Reset
+    | Top of CounterAction
+    | Bottom of CounterAction
+
+let nestedUpdate model action = 
+    let mapCounterAction tag action =
```
Comment (MangelMaxime) [inline samples/browser/virtualdom/virtualdom.fsx]: This function is declared but never used.

File: samples/browser/virtualdom/virtualdom.fsx [hunk 2]
```diff
@@ -56,16 +56,16 @@ type CounterAction =
...
     else window.setTimeout((fun _ -> h (message)), 2000) |> ignore
 
-let counterUpdate model action =
-    match action with
+let counterUpdate model command =
+    match command with
     | Decrement x -> model - x
     | Increment x -> model + x
-    |> (fun m -> m,[],[fakeAjaxCall model])
+    |> (fun m -> m, (fakeAjaxCall model) |> action)
```
Comment (MangelMaxime) [inline samples/browser/virtualdom/virtualdom.fsx]: If I understand well, this `action` function is a wrapper to transform the left into a list of action.

I would change the named of the function to something like: `toActionList`
Just to understand what we are doing with this function.

Comment (mastoj) [inline samples/browser/virtualdom/virtualdom.fsx]: Good comment, changed to `toActionList`.

File: samples/browser/virtualdom/virtualdom.fsx (hunk 1 above)
Comment (mastoj) [inline samples/browser/virtualdom/virtualdom.fsx]: Nice catch, updated the PR and removed the function.

**Comment by alfonsogarciacaro** [comment]
Merged!

**Pull request merged**